src/array.ts

```diff
@@ -355,6 +355,32 @@ export const replaceOrAppend = <T>(
   return [...list, newItem]
 }
 
+/**
+ * If the item matching the condition already exists
+ * in the list it will be removed. If it does not it
+ * will be added.
+ */
+export const toggle = <T>(
+  list: readonly T[],
+  newItem: T,
+  toValue?: null | ((a: T, idx: number) => number | string | symbol),
+  options?: {
+    strategy?: 'prepend' | 'append'
+  }
+) => {
+  if (!list && !newItem) return []
+  if (!list) return [newItem]
+  if (!newItem) return [...list]
+  const matcher = toValue
+    ? (x: T, idx: number) => toValue(x, idx) === toValue(newItem, idx)
+    : (x: T) => x === newItem
+  const existing = list.find(matcher)
+  if (existing) return list.filter((x, idx) => !matcher(x, idx))
+  const strategy = options?.strategy ?? 'append'
+  if (!strategy || strategy === 'append') return [...list, newItem]
```

Comment:
I guess it's a safety check, but why would we need the `!strategy` statement here? Shouldn't it be always defined?

Reply:
Great catch! I just pushed a commit removing the unneeded check 👍